lib/interfaces/typeorm-options.interface.ts

```diff
@@ -47,7 +47,7 @@ export type TypeOrmDataSourceFactory = (
 
 export interface TypeOrmModuleAsyncOptions
   extends Pick<ModuleMetadata, 'imports'> {
-  name?: string;
+  name: string;
```

Comment:
If `name` isn't mentioned, then `default` is used instead. If we make name required, as you propose we'd be making a breaking change.